### DISCUSSION TITLE
How to iterate over a map of `hist`s?

### DISCUSSION THREAD

**Discussion by thediveo** [bpftrace/bpftrace#3366] [Q&A] [answered]
I'm collecting execution times for something or other, ending up with:

```
@histograms[$mykey] = hist($elapsed);
```

Is there a way to iterate over my `@histograms` for the `$mykey` to get the individual histograms one after another?

**Comment by jordalgo** [comment, marked as answer]
Sadly, no. `hist`, `lhist`, and `stats` maps are the only ones you can't iterate over because they have special keys and values - and we just haven't added the plumbing for it yet. The best thing to do at this point would be to just create separate maps but that might be a lot of boilerplate.
